File: src/int80h/misc.c
#include <os/int80/misc.h>
#include <os/kernel.h>

void *int80h_command19_reboot(struct interrupt_frame *frame){
    reboot();
    return 0;
}

void *int80h_command20_shutdown(struct interrupt_frame *frame){
    shutdown();
    return 0;
}

File: src/rust/src/interrupts/idt80/misc.rs
use crate::{
    interrupts::idt::InterruptFrame,
    utils::{reboot, shutdown},
};

pub fn int80h_command19_reboot(_frame: &InterruptFrame) -> u32 {
    reboot();
    0
}

pub fn int80h_command20_shutdown(_frame: &InterruptFrame) -> u32 {
    shutdown();
    0
}
